addons/website_sale/models/sale_order.py

```diff
@@ -295,7 +295,8 @@ def _cart_accessories(self):
                 accessory_products |= line.product_id.accessory_product_ids.filtered(lambda product:
                     product.website_published and
                     product not in products and
-                    product._is_variant_possible(parent_combination=combination)
+                    product._is_variant_possible(parent_combination=combination) and
+                    (product.company_id == line.company_id or not product.company_id)
```

Comment:
Don't change it for this PR but IMO this is better written 
```python
(product.company_id in (line.company_id, False))
```
(Can also be applied in xml)